### ISSUE TITLE
Missing parameter name after comptime arg asserts

### ISSUE THREAD

**Issue by Hejsil** [ziglang/zig#1209]
```
pub fn a(comptime T: type, T) void {}

test "" {
    a(u8, 2);
}
```

```
zig/src/analyze.cpp:3448: VariableTableEntry* add_variable(CodeGen*, AstNode*, Scope*, Buf*, bool, ConstExprValue*, Tld*): Assertion `name' failed.
[1]    21470 abort (core dumped)  zig test test.zig
```

**Comment by Hejsil** [comment]
Related:
It seems that only generic functions have parameters available right after they have been analyzed, while normal functions do not allow this. Is this intended?
```
// error: use of undeclared identifier 'b'
pub fn a(b: u8, c: @typeOf(b)) void {}

test "" {
    a(2, 2);
}
```
```
// Works, even though b is not generic
pub fn a(comptime Q: type, b: u8, c: @typeOf(b)) void {}

test "" {
    a(u8, 2, 2);
}
```

**Comment by andrewrk** [comment]
> Is this intended?

Yes, that is intended. (However, the assertion is not intended, of course.)

It could be made to work. Is there some compelling reason you would want to use `@typeOf` instead of repeating the type?

**Comment by Hejsil** [comment]
@andrewrk I don't really have a usecase other than consistency, but I don't think it matters to much. Was just making sure that was how it should work.

**Comment by andrewrk** [comment]
fixed in fd575fe1f3b45806f2cf823a2abe3727d381d4ed
